### PR TITLE
Add 'wave like water' waving type.

### PR DESCRIPTION
**Pull request by t0ny2**
Adds a new waving type of 3 (wave node like water). Only nodes
with this waving type will be affected by the waving water shader.
Allows water to wave while lava does not. (Issue #5694 )

### CODE REVIEW

**Comment by paramat** [comment]
Yes this should be done instead of hardcoding all liquids to wave.
Requested here https://github.com/minetest/minetest/issues/5694#issuecomment-298760861

**Comment by paramat** [comment]
Why closed? This is needed.

**Comment by paramat** [comment]
If you prefer that someone else takes over let us know.

**Comment by paramat** [comment]
@numberZero this is a very desirable feature, perhaps you could look at the code to see if the implementation is good? If it is i can adopt this.

File: src/client/tile.h [hunk 1]
```diff
@@ -159,7 +159,10 @@ enum MaterialType{
 	TILE_MATERIAL_LIQUID_TRANSPARENT,
 	TILE_MATERIAL_LIQUID_OPAQUE,
 	TILE_MATERIAL_WAVING_LEAVES,
-	TILE_MATERIAL_WAVING_PLANTS
+	TILE_MATERIAL_WAVING_PLANTS,
+	TILE_MATERIAL_WAVING_WATER_BASIC,
+	TILE_MATERIAL_WAVING_WATER_TRANSPARENT,
+	TILE_MATERIAL_WAVING_WATER_OPAQUE
```
Comment (numberZero) [inline src/client/tile.h]: Missing trailing comma (yes it is allowed, and actually preferable for neater future diffs)

File: src/nodedef.cpp [hunk 2]
```diff
@@ -755,6 +755,8 @@ void ContentFeatures::updateTextures(ITextureSource *tsrc, IShaderSource *shdsrc
 			material_type = TILE_MATERIAL_WAVING_PLANTS;
 		else if (waving == 2)
 			material_type = TILE_MATERIAL_WAVING_LEAVES;
+		else if (waving == 3)
+			material_type = TILE_MATERIAL_WAVING_WATER_BASIC;
```
Comment (numberZero) [inline src/nodedef.cpp]: switch

File: src/shader.cpp [hunk 3]
```diff
@@ -646,10 +649,13 @@ ShaderInfo generate_shader(const std::string &name, u8 material_type, u8 drawtyp
 		"TILE_MATERIAL_LIQUID_TRANSPARENT",
 		"TILE_MATERIAL_LIQUID_OPAQUE",
 		"TILE_MATERIAL_WAVING_LEAVES",
-		"TILE_MATERIAL_WAVING_PLANTS"
+		"TILE_MATERIAL_WAVING_PLANTS",
+		"TILE_MATERIAL_WAVING_WATER_BASIC",
+		"TILE_MATERIAL_WAVING_WATER_TRANSPARENT",
+		"TILE_MATERIAL_WAVING_WATER_OPAQUE"
```
Comment (numberZero) [inline src/shader.cpp]: missing comma

File: src/shader.cpp [hunk 4]
```diff
@@ -646,10 +649,13 @@ ShaderInfo generate_shader(const std::string &name, u8 material_type, u8 drawtyp
...
 	};
 
-	for (int i = 0; i < 6; i++){
+	for (int i = 0; i < 9; i++){
```
Comment (numberZero) [inline src/shader.cpp]: ARRLEN is preferable.

**Comment by paramat** [comment]
Thanks i'll adopt this.

**Comment by paramat** [comment]
Now adopting this.

File: src/nodedef.cpp [hunk 5]
```diff
@@ -769,8 +771,13 @@ void ContentFeatures::updateTextures(ITextureSource *tsrc, IShaderSource *shdsrc
 		correctAlpha(tdef, 6);
 		correctAlpha(tdef_overlay, 6);
 		correctAlpha(tdef_spec, CF_SPECIAL_COUNT);
-		material_type = (alpha == 255) ?
-			TILE_MATERIAL_LIQUID_OPAQUE : TILE_MATERIAL_LIQUID_TRANSPARENT;
+
+		if (waving >= 1)
```
Comment (paramat) [inline src/nodedef.cpp]: `if (waving == 3)`
?

**Pull request closed without merging**